packages/nextjs/utils/__test__/contract.test.ts

```diff
@@ -0,0 +1,332 @@
+import { describe, it, expect, beforeEach, vi } from "vitest";
+import {
+  deepMergeContracts,
+  getFunctionsByStateMutability,
+  contracts,
+  parseParamWithType,
+  parseTuple,
+} from "../scaffold-stark/contract";
+import { Abi } from "abi-wan-kanabi";
+
+vi.mock("../contract", () => {
+  const originalModule = vi.importActual("../scaffold-stark/contract");
+  return {
+    ...originalModule,
+    // ...vi.importActual('../scaffold-stark/contract'),
+    parseParamWithType: vi.fn(),
+    encodeParamsWithType: vi.fn(),
+    decodeParamsWithType: vi.fn(),
+  };
+});
```

Comment:
I don't think you should mock the stuff here, if you need to mock the functions on a one off basis, you can use the spy functions of vitest (https://vitest.dev/guide/mocking#functions). 

You can either play with the input or use the spy on the above to produce different cases, if some functions are private, sometimes it implies that you need to treat the logic inside as if they are part of the function without mocking.